tests/Unit/EmogrifierTest.php

```diff
@@ -737,6 +737,198 @@ public function emogrifyNotAppliesCssToNonMatchingElements($css, $expectedHtml)
         static::assertContains(sprintf($expectedHtml, $cssDeclaration1, $cssDeclaration2), $result);
     }
 
+    /**
+     * Provides data to test the following selector specificity ordering:
+     *     * < t < 99t < . < .+t < .+99t < 99. < 99.+t < 99.+99t
+     *     < # < #+t < #+99t < #+. < #+.+t < #+.+99t < #+99. < #+99.+t < #+99.+99t
+     *     < 99# < 99#+t < 99#+99t < 99#+. < 99#+.+t < 99#+.+99t < 99#+99. < 99#+99.+t < 99#+99.+99t
+     * where '*' is the universal selector, 't' is a type selector, '.' is a class selector, and '#' is an ID selector.
+     *
+     * Up to 99 selectors of each type specificity are supported (without making the comparator more complex).
+     *
+     * Specificity ordering for selectors involving pseudo-classes, attributes and `:not` is covered through the
+     * combination of these tests and the equal specificity tests and thus does not require explicit separate testing.
+     *
+     * @return string[][]
+     */
+    public function differentCssSelectorSpecificityDataProvider()
+    {
+        // broken: testing selectors with 99 types or 99 IDs requires support for chaining `:not(h1):not(h1)...` or
+        // `#p4#p4...`, so currently 2 types and/or 2 IDs are tested as 'many' instead of 99 of each.
+        $classP4Repeated = str_repeat('.p-4', 99);
```

Comment:
By the way, please let's reduce the 99 to the minimal number that's relevant for the effect. My educated guess is that it's 2 or 3 (as this is not a performance stress test).

Reply:
I've changed this from 99 to 2 in these tests, as testing that 'many' is more specific than one is sufficient.

To confirm the increase in selector type weighting I've added a single separate dataset with 99.